dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,303 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Core/Field.h>
+#include <DataStreams/StringStreamBlockInputStream.h>
+#include <Debug/DAGProperties.h>
+#include <Debug/MockTiDB.h>
+#include <Debug/dbgKVStore/dbgFuncInvestigator.h>
+#include <Debug/dbgKVStore/dbgRegion.h>
+#include <Debug/dbgTools.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/InterpreterCreateQuery.h>
+#include <Interpreters/InterpreterSelectQuery.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTSelectQuery.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/ParserSelectQuery.h>
+#include <Parsers/parseQuery.h>
+#include <Storages/IManageableStorage.h>
+#include <Storages/KVStore/KVStore.h>
+#include <Storages/KVStore/MultiRaft/RegionData.h>
+#include <Storages/KVStore/MultiRaft/RegionRangeKeys.h>
+#include <Storages/KVStore/Region.h>
+#include <Storages/KVStore/TMTContext.h>
+#include <TiDB/Schema/TiDBSchemaManager.h>
+
+#include <unordered_map>
+#include <unordered_set>
+
+namespace DB
+{
+
+struct MatchResult
+{
+    String mapped_database_name;
+    String table_name;
+    std::vector<std::pair<UInt64, UInt64>> in_default;
+    std::vector<std::pair<UInt64, UInt64>> in_write;
+    std::vector<UInt64> in_lock;
+    std::unordered_map<UInt64, RegionPtr> regions;
+
+    std::string toString() const
+    {
+        FmtBuffer fmt_buf;
+        fmt_buf.fmtAppend("default_cf ");
+        fmt_buf.joinStr(
+            in_default.begin(),
+            in_default.end(),
+            [](const auto & a, const auto &) { return fmt::format("{}-{}", a.first, a.second); },
+            ":");
+        fmt_buf.fmtAppend("; write_cf ");
+        fmt_buf.joinStr(
+            in_write.begin(),
+            in_write.end(),
+            [](const auto & a, const auto &) { return fmt::format("{}-{}", a.first, a.second); },
+            ":");
+        fmt_buf.fmtAppend("; lock_cf ");
+        fmt_buf.joinStr(in_lock.begin(), in_lock.end(), ":");
+        for (const auto & [region_id, region] : regions)
+        {
+            fmt_buf.fmtAppend(
+                "; region {} {}, tikv_range: {}; ",
+                region_id,
+                region->getDebugString(),
+                region->getRange()->toDebugString());
+        }
+        return fmt_buf.toString();
+    }
+};
+
+/// 1. If the arg is [start1, end1], changed to 123
```

Comment:
Describe the behavior instead of linking to the code line

Reply:
This is curious, seem that my submitted code is reverted...

Reply:
fixed